src/read.jl

```diff
@@ -183,16 +191,32 @@ function read_demand_flexibility(filepath, interval)::DemandFlexibility
         # Only try loading the profiles if demand flexibility is enabled
         if demand_flexibility["enabled"] == "not_specified" ||
             (demand_flexibility["enabled"])
-            # Try loading the demand flexibility profiles
-            try
-                demand_flexibility["flex_amt_" * s] = DataFrames.DataFrame(
-                    CSV.File(joinpath(filepath, "demand_flexibility_" * s * ".csv"))
-                )
-                println("...loading demand flexibility " * s * " profiles")
-            catch e
-                println("Demand flexibility " * s * " profile not found in " * filepath)
+            # Try loading the demand flexibility profile from user defined csvs
+            if demand_flexibility["source"] == "user"
+                try
+                    demand_flexibility["flex_amt_" * s] = DataFrames.DataFrame(
+                        CSV.File(joinpath(filepath, "demand_flexibility_" * s * ".csv"))
+                    )
+                    println("...loading demand flexibility " * s * " profiles")
+                catch e
+                    println("Demand flexibility " * s * " profile not found in " * filepath)
+                end
+            # Load from DOE dataset if selected
+            elseif demand_flexibility["source"] == "doe"
+                try
+                    demand_flexibility["flex_amt_" * s] = DataFrames.DataFrame(
+                        CSV.File(joinpath(filepath, "doe_flex_2016.csv"))
+                    )
+                    println("...loading DOE demand flexibility " * s * " profiles")
+                catch e
+                    println("Demand flexibility " * s * " profile not found in " * filepath)
+                end               
+            # throw an error otherwise
+            else
+                demand_flexibility["flex_amt_" * s] = nothing
+                println("Demand flexibility parameter" * demand_flexibility["source"] *
+                        "is invalid!")
```

Comment:
In this case, might it be best to throw an error that stops the simulation? Perhaps by doing something like `throw(ErrorException("The provided demand flexibility source is invalid"))`? Ideally, `PowerSimData` will (eventually) catch something like this, but it's always good for the standalone version of `REISE.jl` to catch this as well.